fcli-core/fcli-aviator/src/main/java/com/fortify/cli/aviator/_common/session/user/cli/cmd/AviatorUserSessionLoginCommand.java

```
@@ -44,6 +56,31 @@ protected void logoutBeforeNewLogin(String sessionName, AviatorUserSessionDescri
     protected AviatorUserSessionDescriptor login(String sessionName) {
         String resolvedToken = tokenResolver.getToken();
         Date expiryDate = AviatorJwtUtils.extractExpiryDateFromToken(resolvedToken);
+        String tenantName = AviatorJwtUtils.extractTenantNameFromToken(resolvedToken);
+
+        LOG.info("Default Aviator admin configuration found. Attempting to validate user token...");
+        try (AviatorGrpcClient client = AviatorGrpcClientHelper.createClient(sessionLoginOptions.getAviatorUrl())) {
+
+            TokenValidationResponse validationResponse = client.validateUserToken(resolvedToken, tenantName);
+
+            if (!validationResponse.getValid()) {
+                String errorMsg = validationResponse.getErrorMessage();
+                String fullError = "Aviator user token validation failed: " +
+                        (errorMsg == null || errorMsg.isBlank() ? "Token is invalid." : errorMsg);
+                throw new AviatorSimpleException(fullError);
+            }
+            LOG.info("Aviator user token validated successfully with the Aviator server.");
+        } catch (AviatorSimpleException e) {
+            throw e;
+        } catch (Exception e) {
+            LOG.warn("An error occurred while trying to validate the Aviator user token with the server using the 'default' admin config: {}. " +
```

What's the rationale for logging a warning, instead of throwing an exception? Usually (for SSC/ScanCentral/FoD), we throw an exception if connection check fails, which will result in a non-zero exit code to inform the caller process like CI pipeline about this failure.

---

This was intended to handle cases like using an older (e.g. 23.2) Aviator server with a newer FCLI version, where validation would always fail but you’re right and i forgot that scenario will actually never happen and I forgot that. I agree I’ll update it to fail fast so it aligns better with our CI/CD expectations.

---

@kireetivar I understand, and in fact having these as warnings allowed me to verify the issue with the static field mentioned below (testing these 25.3 updates against 25.2 Aviator service). Backward compatibility with older Aviator backend versions is also something to consider once we allow customers to run their own Aviator service I guess, so certainly something to keep in mind.

When testing this PR, I think the warning showed something like 'endpoint doesn't exist', so apparently we can differentiate that from other error conditions. So, *if* we'd want to have backward compatibility, we could potentially check for this specific error to identify whether we're talking to Aviator 25.2 or 25.3, and then take the appropriate action (i.e., just ignore this particular error, or output a warning stating that connection check is not supported by current Aviator backend version).